pkg/controller/job/job_controller.go

```diff
@@ -1662,6 +1662,11 @@ func activePodsForRemoval(job *batch.Job, pods []*v1.Pod, rmAtLeast int) []*v1.P
 		left = pods
```

Comment:
How about the expression above: `left = make([]*v1.Pod, 0, len(pods)-rmAtLeast)` for indexed jobs. Can this also suffer in a similar way, when `rmAtLeast=4` and `len(pods)=1` it would evaluate to negative value. If so, please fix as it looks related and add a unit test.